src/coreclr/vm/methodtablebuilder.cpp

```diff
@@ -4311,6 +4313,9 @@ VOID    MethodTableBuilder::InitializeFieldDescs(FieldDesc *pFieldDescList,
                   pszFieldName
                   );
 
+        if (pLayoutFieldInfo)
+            accumulatedSize += pLayoutFieldInfo->m_placement.m_size;
```

Comment:
It is suspicious that this does not take alignment into account and that the check for overflow is not right here.

I think the fix should be structured more like the fix that Michal pushed for the managed type system. Always check for overflow right before the (existing) calculation or assignment would actually overflow. It should not require adding new ways to compute sizes.

Reply:
Inside `InitializeFieldDescs`, alignment didn't seem to be taken into consideration. I looked around and found the following place, where it is the size of the Sequential struct. So I added a check there to see it overflows `int`.

https://github.com/dotnet/runtime/blob/a86987ccab917433d065fe5dc8870fc261f79d14/src/coreclr/vm/methodtablebuilder.cpp#L1765